### PR TITLE
ACAS-98: Add test for case-insensitive bulk load and routes for cmpdRegAdmin categoricals

### PR DESCRIPTION
**Pull request by bffrost**
## Description
- Added get & create methods for vendor, scientist, stereo category, salt, and physical state, along with tests
- Added a dry_run flag to `register_sdf` that will trigger validation if set to `True`
- Added a test with wrong-case categorical values to test ACAS-98 making those case-insensitive. This will fail until associated ACAS and Roo changes are merged.

## How Has This Been Tested?
Ran acasclient unit tests locally

### CODE REVIEW

**Comment by brianbolt** [comment]
NM. I think I get why that's an issue.  Matches "Bob" as "bob" but still save "Bob" in to the table.

File: tests/test_acasclient.py [hunk 1]
```diff
@@ -1797,4 +1797,107 @@ def test_044_author_and_role_apis(self):
...
+        STEREO_CATEGORY = 'Unknown'
+        SALT_ABBREV = 'HCl'
+        SALT_MOL = "\n  Ketcher 05182214202D 1   1.00000     0.00000     0\n\n  1  0  0     1  0            999 V2000\n    6.9500   -4.3250    0.0000 Cl  0  0  0  0  0  0  0  0  0  0  0  0\nM  END\n"
+        PHYSICAL_STATE = 'plasma'
+        VENDOR = 'ThermoFisher'
+        # TODO Lot Chemist, Stereo Category, Salt Abbrev, Physical State, Vendor
+        # Get Lot Chemists
+        lot_chemists = self.client.get_cmpdreg_scientists()
+        # Create Lot Chemist
+        if CHEMIST not in [c['code'] for c in lot_chemists]:
```
[the remark this answers is not in the document]
Comment (bffrost) [reply tests/test_acasclient.py:1816]: I don't think it should. This line will see `bob` in the list of codes for saved chemists and will skip creating it again.

[the remark this answers is not in the document]
Comment (iamanandkr) [reply tests/test_acasclient.py:1816]: Ahh, I should have mentioned the full query in the original comment 🙈 . Do we want this i.e if I break the API later will the tests not tell me about it? 🤔

[the remark this answers is not in the document]
Comment (bffrost) [reply tests/test_acasclient.py]: Good point. I've done some more work here and created a separate CRUD test

**Review by brianbolt — APPROVED** [review]
LGTM

**Pull request merged**